src/utils/themes-manager.js

```diff
@@ -0,0 +1,79 @@
+//@flow
+
+import {hexToHsl} from './color-format-convertors';
+import {style} from '../index';
+
+const cssVarNames: ThemesManagerConfig = {
+  colors: {
+    primary: '--playkit-primary-hsl-hue',
+    secondary: '--playkit-secondary-hsl-hue',
+    success: '--playkit-success-hsl-hue',
+    danger: '--playkit-danger-hsl-hue',
+    warning: '--playkit-warning-hsl-hue',
+    live: '--playkit-live-color',
+    playerBackground: '--playkit-player-background-color'
+  }
+};
+
+const dynamicColoredIconsSvgUrlVars = [
+  '--playkit-icon-check-active-url',
+  '--playkit-icon-data-url',
+  '--playkit-icon-chromecast-url',
+  '--playkit-icon-quality-HD-active-url',
+  '--playkit-icon-quality-4K-active-url',
+  '--playkit-icon-quality-8K-active-url'
+];
+
+// eslint-disable-next-line require-jsdoc
+export class ThemesManager {
+  player: Object;
+  config: Object;
+
+  // eslint-disable-next-line require-jsdoc
+  constructor(player: Object, config: UserTheme) {
+    this.player = player;
+    this.config = config;
+  }
+
+  // eslint-disable-next-line require-jsdoc
+  applyUserTheme(): void {
+    if (this.config) {
+      this.setColors(this.config);
+    }
+  }
+
+  // eslint-disable-next-line require-jsdoc
+  getHueComponent(hex: string): number {
+    const hsl = hexToHsl(hex);
+    return hsl[0];
+  }
+
+  // eslint-disable-next-line require-jsdoc
+  setColors(config: ThemesManagerConfig): void {
+    if (config.colors.primary) {
+      // this.setColor(cssVarNames.colors.primary, config.colors.primary);
+      this.setSvgFillColor(config.colors.primary);
+    }
+
+    for (const color in config.colors) {
+      this.setColor(cssVarNames.colors[color], config.colors[color]);
+    }
+  }
+
+  // eslint-disable-next-line require-jsdoc
+  setColor(cssVarName: string, color: string): void {
+    const hue = this.getHueComponent(color);
+    document.querySelector<HTMLElement>('.playkit-player').style.setProperty(cssVarName, `${hue}deg`);
```

Comment:
this query will return every player on the page, if theres more than 1. meaning you'll also be setting the style for every player on the page.

Reply:
document.querySelector returns only one element (the first within the document that matches the specified selector) but anyway the issue is still there that it will return the wrong player so what do you suggest how to identify the specific player ?

Reply:
you can use something like this for example
https://github.com/kaltura/playkit-js-visibility/blob/1e125cbc275e95918a7cc9ac5b31d5501baa5cd4/src/visibility.js#L96